### PR TITLE
Jest coverage

### PR DESCRIPTION
**Pull request by clarkbw**
Associated Issue: #2319

### Summary of Changes

* add `test-coverage` script
* add coverage reporting to circleci and upload to codecov

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -27,6 +27,7 @@
     "lint-md": "remark -q *.md src/**/*.md configs/*.md docs/**/*.md",
     "lint-fix": "yarn lint-js -- --fix",
     "test": "jest --forceExit",
+    "test-coverage": "yarn test -- --coverage",
```
Comment (clarkbw) [inline package.json]: I wasn't sure if we wanted add coverage to the default test script instead of a creating a new one.

Comment (jasonLaster) [inline package.json]: sure - this is probably fine. we can always update this later when we're using flow as well

**Comment by jasonLaster** [comment]
Should we see codecov here?

**Comment by clarkbw** [comment]
> Should we see codecov here?

Yes, but I turned it off previously since we weren't getting flow coverage.  Just turned it back on now.  Maybe if I close and reopen this it'll happen...

**Comment by clarkbw** [comment]
Adding a fix for the lint issue that I believe came from #2420

**Comment by jasonLaster** [comment]
ahh - thanks

File: src/components/shared/Accordion.css [hunk 2]
```diff
@@ -60,6 +60,6 @@
   height: 16px;
 }
 
-accordion .header-buttons button::-moz-focus-inner {        
+.accordion .header-buttons button::-moz-focus-inner {
```
Comment (clarkbw) [inline src/components/shared/Accordion.css]: I also added a `.` here which is what I believe is needed and was just missed from copy / pasta

**Comment by clarkbw** [comment]
Coverage is now being reported in CI.

screen shot 2017-03-23 at 10 42 15 am

And I'm seeing the JSON artifact: https://400-55521579-gh.circle-artifacts.com/0/tmp/circle-junit.y1G4yTz/jest-coverage/coverage-final.json

**Comment by montogeek** [comment]
Cool!

**Pull request merged**